mate-panel/applet.c

```diff
@@ -1135,18 +1140,32 @@ mate_panel_applet_save_position (AppletInfo *applet_info,
 	if (locked) {
 		/* Until position calculations are refactored to fix the issue of the panel applets
 		   getting reordered on resolution changes...
-		   .. don't save position/right-stick on locked applets */
+		   .. don't save position/relative-to-edge on locked applets */
 		return;
 	}
 
-	right_stick = panel_is_applet_right_stick (applet_info->widget) ? 1 : 0;
+	edge_relativity = panel_determine_applet_edge_relativity (applet_info->widget);
+	if (g_settings_is_writable (applet_info->settings, PANEL_OBJECT_RELATIVE_TO_EDGE_KEY) &&
+	    g_settings_get_enum (applet_info->settings, PANEL_OBJECT_RELATIVE_TO_EDGE_KEY) != edge_relativity)
+		g_settings_set_enum (applet_info->settings, PANEL_OBJECT_RELATIVE_TO_EDGE_KEY, edge_relativity);
+
+	/*
+	 * If the deprecated `panel-right-stick` setting is set, unset it.
+	 * See the `panel_profile_load_object ()` function in the file
+	 * `panel-profile.c` for a lengthy comment detailing why this is
+	 * done.
+	 */
 	if (g_settings_is_writable (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY) &&
-	    (g_settings_get_boolean (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY) ? 1 : 0) != right_stick)
-		g_settings_set_boolean (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY, right_stick);
+	    g_settings_get_boolean (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY) != FALSE)
```

Comment:
please, use
```
g_settings_get_boolean (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY)
```
instead of
```
g_settings_get_boolean (applet_info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY) != FALSE)
```
for the if-clause